### ISSUE TITLE
e2e flaky: Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slow] Deployment Should scale from 1 pod to 3 pods and from 3 to 5 and verify decision stability

### ISSUE THREAD

**Issue by dchen1107** [kubernetes/kubernetes#23536]
Jenkins build kubernetes-e2e-gke-subnet/2545/ failed:

```
Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slow] Deployment Should scale from 1 pod to 3 pods and from 3 to 5 and verify decision stability

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/horizontal_pod_autoscaling.go:50 Mar 28 02:53:23.996: Number of replicas has changed: expected 3, got 4


Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slow] ReplicaSet Should scale from 1 pod to 3 pods and from 3 to 5 and verify decision stability

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/horizontal_pod_autoscaling.go:61 Mar 28 03:28:13.740: Number of replicas has changed: expected 3, got 4
```

Related logs can be accessed at: https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gke-subnet/2545/

**Comment by roberthbailey** [comment]
This is flaky in the 1.2 release branch as well. Depending on what the underlying issue is, we may want/need to cherry pick a fix for a 1.2 patch release.

**Comment by roberthbailey** [comment]
I'm also seeing similar flakes on the `Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slow] ReplicaSet Should scale from 1 pod to 3 pods and from 3 to 5 and verify decision stability` and `Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slow] ReplicationController Should scale from 1 pod to 3 pods and from 3 to 5 and verify decision stability` test cases.

**Comment by piosz** [comment]
#21664 should help, @jszczepkowski is working on it

**Comment by jszczepkowski** [comment]
Should be fixed by #25726.
